### ISSUE TITLE
Getting Body size 0

### ISSUE THREAD

**Issue by grafael** [go-resty/resty#300]
For some **POST** requests I'm getting Body length = 0 error. 
I did the exact same request using curl and worked properly, also I did use pure **http.request** and worked. I have no idea what else I could configure to get that working.

Snippet:

```golang
data := `{"name":"test"}`
resp, err := c.client.New().R().
	SetHeader("Content-Type", "application/json").
	SetBody(data).
	Post(url)
```
 
http: ContentLength=181 with Body length 0

```
Error      : Post <my_endpoint_here>: http: ContentLength=181 with Body length 0
Status Code: 0
Status     :
Time       : 175.825µs
Received At: 2020-01-24 08:56:39.710017 -0300 -03 m=+301.171273859
Body       :

Params     :
 map[]

Request Trace Info:
DNSLookup    : 31.906079ms
ConnTime     : 6.764µs
TLSHandshake : 335.204955ms
ServerTime   : -189.941µs
ResponseTime : 319.94µs
TotalTime    : 175.825µs
IsConnReused : true
IsConnWasIdle: true
ConnIdleTime : 96.155µs
```

**Comment by jeevatkm** [comment]
@grafael Would you mind providing an example program to demo the issue?

**Comment by jeevatkm** [comment]
@grafael Any updates? I think you could also explore the resty test cases to validate the resty.

**Comment by wjh000123** [comment]
@jeevatkm this issue occurred when the reponse is in HTTP status - 307.

**Comment by manojkva** [comment]
I had this same issue while using resty v2.2.0 with Gin framework. Resty was adding up an additional "/" at the end of the Endpoint even when not provided. 
This was not allowed in Gin i suppose (https://github.com/gin-gonic/gin/issues/1004).
I got it resolved by downgrading my resty version to v2.0.0 .
Please note that v2.1.0 too had the same issue.

**Comment by jeevatkm** [comment]
@wjh000123  @manojkva Thanks for the inputs, will have a look.

**Comment by alexio777** [comment]
I have exactly the same problem.

`	resp, err := hc.rc.R().
		SetHeader("Content-Type", "application/octet-stream").
		SetBody(data).
		Post(fmt.Sprintf("%s%s", gRemoteAPIURL, path))
`
where data is []byte

and I get an error: http: ContentLength=224 with Body length 0

**Comment by moorereason** [comment]
@alexio777, can you provide a complete test program to demonstrates the issue?

**Comment by jeevatkm** [comment]
@grafael @alexio777 I have tried to reproduce it. I am unable to. Could you please provide an end-to-end unit test case to showcase an issue?

**Comment by Hyingerrr** [comment]
I also encountered such problems in production. Is there any solution? thinks.

**Comment by jeevatkm** [comment]
Fixed by PR #380

**Comment by albttx** [comment]
I had the same issues, my probleme was that i initialize my "Service" with a new request for always having  the http header

eg: 

```go
func NewService(cfg Config) (Service, error) {
	c := &Client{
		gqlEndpoint: cfg.HasuraEndpoint,
	}

	c.gqlClient = resty.New().
		R().
		SetHeader("x-hasura-admin-secret", cfg.HasuraAdminSecret)
	return c, nil
}
```
creating a new `R()` on each request fixed the issue
